solr/core/src/java/org/apache/solr/core/SolrXmlConfig.java

```diff
@@ -327,6 +327,9 @@ private static NodeConfig fillSolrSection(NodeConfig.NodeConfigBuilder builder,
         case "sharedLib":
           builder.setSharedLibDirectory(value);
           break;
+        case "modules":
+          builder.setModules(value);
```

Comment:
Did not do that for sharedLib, so chose to keep it transparent with user input here too. Currently the split and strip logic is done by consumer ModuleUtils, but it could of course have been moved earlier. Guess I prefer to not have a ton of additional logig in plain getters/setters.